x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list.tsx

```diff
@@ -8,6 +8,7 @@
 import React, { useEffect, useState } from 'react';
 
 import { useActions, useValues } from 'kea';
+import useThrottle from 'react-use/lib/useThrottle';
```

Comment:
TIL, this is great and great timing as I needed this today.

Reply:
Note for the future: it's better and more idiomatic to put throttling in the Engines List logic file so it's shared across every instance where you may use this action. You can use the `breakpoint` feature in Kea to do this, like here:
https://github.com/elastic/kibana/tree/main/x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/source_logic.ts